axum-extra/src/extract/multipart.rs

```diff
@@ -246,6 +247,58 @@ impl MultipartError {
     fn from_multer(multer: multer::Error) -> Self {
         Self { source: multer }
     }
+
+    /// Get the response body text used for this rejection.
+    pub fn body_text(&self) -> String {
+        self.source.to_string()
+    }
+
+    /// Get the status code used for this rejection.
+    pub fn status(&self) -> http::StatusCode {
+        status_code_from_multer_error(&self.source)
+    }
+}
+
+fn status_code_from_multer_error(err: &multer::Error) -> StatusCode {
+    match err {
+        multer::Error::UnknownField { .. }
+        | multer::Error::IncompleteFieldData { .. }
+        | multer::Error::IncompleteHeaders
+        | multer::Error::ReadHeaderFailed(..)
+        | multer::Error::DecodeHeaderName { .. }
+        | multer::Error::DecodeContentType(..)
+        | multer::Error::NoBoundary
+        | multer::Error::DecodeHeaderValue { .. }
+        | multer::Error::NoMultipart
+        | multer::Error::IncompleteStream => StatusCode::BAD_REQUEST,
+        multer::Error::FieldSizeExceeded { .. } | multer::Error::StreamSizeExceeded { .. } => {
+            StatusCode::PAYLOAD_TOO_LARGE
+        }
+        multer::Error::LockFailure => StatusCode::INTERNAL_SERVER_ERROR,
+        multer::Error::StreamReadFailed(err) => {
+            if let Some(err) = err.downcast_ref::<multer::Error>() {
+                return status_code_from_multer_error(err);
+            }
```

Comment:
Why would `multer::Error` recursively contain another `multer::Error` for the `StreamReadFailed` variant?

Reply:
@jplatte Because of below chain.

1. https://github.com/rousan/multer-rs/blob/a49e03645a307110691f90493070c2e6ddfa5d00/src/error.rs#L60
1. https://github.com/rousan/multer-rs/blob/a49e03645a307110691f90493070c2e6ddfa5d00/src/field.rs#L342
1. https://github.com/rousan/multer-rs/blob/a49e03645a307110691f90493070c2e6ddfa5d00/src/buffer.rs#L50
1. https://github.com/rousan/multer-rs/blob/a49e03645a307110691f90493070c2e6ddfa5d00/src/multipart.rs#L126
1. https://github.com/hyperium/http-body/blob/6f722a14f7c6488cd53dcc4126a84e6ee201964f/src/limited.rs#L52

So, final error looks like:

```rust
axum::extract::multipart::MultipartError {
    source: multer::Error::StreamReadFailed(
        Box::new(multer::Error::StreamReadFailed(
            Box::new(http_body::LengthLimitError)
        ))
    )
}
```